base/server/src/main/java/com/netscape/cms/realm/PKIRealm.java

```diff
@@ -44,6 +43,19 @@ public class PKIRealm extends RealmBase {
 
     private static Logger signedAuditLogger = SignedAuditLogger.getLogger();
 
+    protected CMSEngine engine;
+
+    public PKIRealm() {
+    }
+
+    public CMSEngine getCMSEngine() {
+        return engine;
+    }
+
+    public void setCMSEngine(CMSEngine engine) {
+        this.engine = engine;
+    }
+
     protected String getName() {
         return "PKIRealm";
     }
```

Comment:
Not part of your diff, but this could be `this.getClass().getSimpleName()` to make it less brittle. The method is not actually used anywhere by the looks of it so it could just be dropped.